### DISCUSSION TITLE
HTML entities no longer rendering

### DISCUSSION THREAD

**Discussion by lorezyra** [hexojs/hexo#5193] [Question / Answer] [answered]
Until recently, [HTML entities](https://html.spec.whatwg.org/multipage/named-characters.html#named-character-references) in my markdown posts were rendering properly when generated. Suddenly, they have stopped rendering and I see `&mdash;` instead of `—`. I'm not sure where to start on troubleshooting the cause as I'm still using node 19 with hexo 6.3. I did recently upgrade `hexo-util`, `hexo-generator-index`, `hexo-generator-archive`, and `hexo-generator-category`. Why would that break rendering HTML entities?

Could someone suggest where I should start troubleshooting this?

## `package.json`

```json
{
  "name": "hexo.blog.RichieBartlett.com",
  "version": "2021.05",
  "private": true,
  "license": "LGPL-3.0-or-later",
  "scripts": {
    "build": "hexo clean && hexo gc -w=100 && hexo generate --debug",
    "calendar": "hexo gc",
    "clean": "hexo clean",
    "deploy": "hexo deploy",
    "server": "hexo clean && hexo server",
    "upgrade": "./upGradeNodeJS.sh",
    "s3": "echo setup S3 plugin",
    "debug": "clear && hexo clean && node --inspect-brk=4000 $(which hexo) server --debug"
  },
  "hexo": {
    "version": "6.3.0"
  },
  "dependencies": {
    "expand-range": "^2.0.2",
    "filename-regex": "^2.0.1",
    "hexo": "^6.3.0",
    "hexo-auto-category": "^0.2.1",
    "hexo-calendar": "^1.0.8",
    "hexo-filter-mathjax": "^0.9.0",
    "hexo-footnote": "^1.0.4",
    "hexo-generator-archive": "^2.0.0",
    "hexo-generator-category": "^2.0.0",
    "hexo-generator-feed": "^3.0.0",
    "hexo-generator-index": "^3.0.0",
    "hexo-generator-readtime": "^1.3.2",
    "hexo-generator-search": "^2.4.3",
    "hexo-generator-seo-friendly-sitemap": "^0.2.1",
    "hexo-generator-tag": "^2.0.0",
    "hexo-helper-crypto": "^1.1.0",
    "hexo-renderer-ejs": "^2.0.0",
    "hexo-renderer-markdown-it": "^6.1.0",
    "hexo-renderer-stylus": "^2.1.0",
    "hexo-server": "^3.0.0",
    "hexo-tag-twitter": "^1.0.2",
    "hexo-tag-wikipedia": "^1.0.1",
    "hexo-tag-youtube-responsive": "^0.4.2",
    "hexo-util": "^3.0.1",
    "is-posix-bracket": "^0.1.1",
    "object.omit": "^3.0.0",
    "parse-glob": "^2.0.1",
    "preserve": "^0.2.0",
    "regex-cache": "^0.4.4"
  },
  "devDependencies": {
    "0x": "^4.1.4",
    "c8": "^7.12.0",
    "eslint": "^8.8.0",
    "eslint-config-hexo": "^5.0.0",
    "heapdump": "^0.3.15",
    "hexo-renderer-marked": "^6.0.0",
    "lint-staged": "^13.0.3"
  },
  "dev": "NODE_OPTIONS='--inspect' hexo --debug",
  "engines": {
    "node": ">=19"
  }
}

```

**Comment by yoshinorin** [comment]
@lorezyra 

Not sure, but I assume caused by [marked](https://marked.js.org/) or [markdown-it](https://markdown-it.github.io/). Did you upgrade `hexo-renderer-marked` or `hexo-renderer-markdown-it`?

**Reply by lorezyra** [reply]
I have Markdown-IT as the renderer in my `package.json` file:

```json
    "hexo-renderer-markdown-it": "^6.1.0",
```

![image](https://user-images.githubusercontent.com/1673222/234811347-8b4f96ad-2bdb-47a2-aa38-9704c539d8e0.png)

Looks [like v7 was released](https://github.com/hexojs/hexo-renderer-markdown-it) a couple months back and I have yet to update to it. Maybe I should?

**Reply by yoshinorin** [reply, marked as answer]
> Looks [like v7 was released](https://github.com/hexojs/hexo-renderer-markdown-it) a couple months back and I have yet to update to it. Maybe I should?

I think you don't need to try it if you haven't updated it recently.

If I were in your position, I would try to roll back `hexo-util` to an older version (before the upgraded version). And I'll check `—` correctory.  I don't think `hexo-generator-index`, `hexo-generator-archive`, and `hexo-generator-category` are related to this issue.

**Reply by lorezyra** [reply]
I've downgraded `hexo-util` to version 2.7.0 and tried to replicate the issue. I can't seem to find the problem. Upgraded it back to 3.0.1, and the problem has disappeared. Not sure what caused my earlier issue, but it appears resolved now.

I can only speculate that the process of downgrading and upgrading may have fixed a corrupted package somewhere. In any case, the problem has apparently resolved itself.

Thanks for your help!! 🙇🏻‍♂️

**Reply by yoshinorin** [reply]
> I've downgraded hexo-util to version 2.7.0 and tried to replicate the issue. I can't seem to find the problem. Upgraded it back to 3.0.1,

Honestly I'm not sure why this problem happens upgrade from `2.7.0` to `3.0.1`.  The main change is the migration from JavaScript to TypeScript 🤔 

> https://github.com/hexojs/hexo-util/compare/2.7.0...v3.0.1

**Comment by lorezyra** [comment]
Okay, I'm seeing this issue again. From what I can see, it's not the markdown renderer at fault. In the DOM, I see `&mdash;` If it was `&amp;mdash;` then I would look closer at the markdown-it renderer. But, it appears this is an issue with my macOS. Both Chrome and Safari are affected. I haven't dug deep enough to see why, but I can now confidently say it has nothing to do with Hexo.
